pkg/apis/serving/v1alpha2/kfservice_status.go

```diff
@@ -53,43 +54,61 @@ func (ss *KFServiceStatus) GetCondition(t apis.ConditionType) *apis.Condition {
 	return conditionSet.Manage(ss).GetCondition(t)
 }
 
-// PropagateDefaultConfigurationStatus propagates the default Configuration status and applies its values
+func (ss *KFServiceStatus) PropagateDefaultStatus(endpoint string, defaultStatus *knservingv1alpha1.ServiceStatus) {
+	switch endpoint {
+	case constants.Predictor:
+		ss.PropagateDefaultPredictorStatus(defaultStatus)
+	case constants.Explainer:
+	case constants.Transformer:
+	}
+}
+
+func (ss *KFServiceStatus) PropagateCanaryStatus(endpoint string, canaryStatus *knservingv1alpha1.ServiceStatus) {
+	switch endpoint {
+	case constants.Predictor:
+		ss.PropagateCanaryPredictorStatus(canaryStatus)
+	case constants.Explainer:
+	case constants.Transformer:
+	}
+}
+
+// PropagateDefaultPredictorStatus propagates the default predictor status and applies its values
 // to the Service status.
-func (ss *KFServiceStatus) PropagateDefaultConfigurationStatus(defaultConfigurationStatus *knservingv1alpha1.ConfigurationStatus) {
-	ss.Default.Name = defaultConfigurationStatus.LatestCreatedRevisionName
-	configurationCondition := defaultConfigurationStatus.GetCondition(knservingv1alpha1.ConfigurationConditionReady)
+func (ss *KFServiceStatus) PropagateDefaultPredictorStatus(defaultStatus *knservingv1alpha1.ServiceStatus) {
+	ss.Default.Name = defaultStatus.LatestCreatedRevisionName
+	serviceCondition := defaultStatus.GetCondition(knservingv1alpha1.ServiceConditionReady)
 
 	switch {
-	case configurationCondition == nil:
-	case configurationCondition.Status == v1.ConditionUnknown:
-		conditionSet.Manage(ss).MarkUnknown(DefaultPredictorReady, configurationCondition.Reason, configurationCondition.Message)
-	case configurationCondition.Status == v1.ConditionTrue:
+	case serviceCondition == nil:
+	case serviceCondition.Status == v1.ConditionUnknown:
+		conditionSet.Manage(ss).MarkUnknown(DefaultPredictorReady, serviceCondition.Reason, serviceCondition.Message)
```

Comment:
@ellis-bigelow Don't we need to propagate Unknown condition too?

Reply:
I'd suggest keep it since it is one of the value for pod condition status https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions, I have  seen this before when not able to contact the node although it is rare.